packages/eslint-plugin-office-addins/src/rules/call-sync-after-load.ts

```diff
@@ -0,0 +1,89 @@
+import { TSESTree } from "@typescript-eslint/experimental-utils";
+import { Variable } from "@typescript-eslint/experimental-utils/dist/ts-eslint-scope";
+import {
+  findPropertiesRead,
+  findReferences,
+  OfficeApiReference,
+} from "../utils";
+
+export = {
+  name: "call-sync-after-load",
+  meta: {
+    type: <"problem" | "suggestion" | "layout">"suggestion",
+    messages: {
+      callSyncAfterLoad:
+        "Call context.sync() after calling load on '{{name}}' property on the '{{loadValue}}' and before reading properties",
+    },
+    docs: {
+      description:
+        "Always call load on an object followed by a context.sync() before reading it or one of its properties.",
+      category: <
+        "Best Practices" | "Stylistic Issues" | "Variables" | "Possible Errors"
+      >"Best Practices",
+      recommended: <false | "error" | "warn">false,
+      url: "https://docs.microsoft.com/en-us/office/dev/add-ins/develop/application-specific-api-model?view=powerpoint-js-1.1#load",
+    },
+    schema: [],
+  },
+  create: function (context: any) {
+    let apiReferences: OfficeApiReference[] = [];
+
+    function findLoadBeforeSync(): void {
+      const needSync: Set<Variable> = new Set<Variable>();
+      const needLoadAndSync: Set<Variable> = new Set<Variable>();
+
+      apiReferences.forEach((apiReference) => {
+        const operation = apiReference.operation;
+        const reference = apiReference.reference;
+        const variable = reference.resolved;
+
+        if (operation === "Write" && variable) {
+          needLoadAndSync.add(variable);
```

Comment:
We don't always have to call load after a write . . . which is what this is making me think . . . you can get and object without loading a property of it.  I think what you mean to track is variables that could have load calls.  But the fact that they are in the apiReference already determined that . . . so I'm not sure why that is needed here.

Reply:
I'm trying to detect the following pattern:
get()* variables (need a write) -> load -> sync -> read

All these steps must exist for the rule to apply successfully. If we don't check for the first get()* variables, we can get false positives if people have a load funcion on their projects

Reply:
If the only reason for the get*() tracking is to verify the load call is legit, then that is already taken care of by  findReferences(context.getScope()) which narrows down the list of references to those that should be from the office api.  You don't need to then track it again when looking for the load -> sync -> read pattern.

Reply:
The only purpose of the get*() is indeed to check if the load call is legit. You are right, I do not need the get* tracking and just removed it.